src/tree/link.rs

```diff
@@ -61,21 +61,51 @@ impl<'a> LinkLocation<'a> {
         let mut link_str = link.as_ref();
 
         // Check for direct URLs or anchor links
-        if is_url(link_str) || link_str.starts_with('#') {
+        // TODO: parse local links into LinkLocation::Page
+        // Known bug: single "/" parsed into Url instead of Page
+        if is_url(link_str) || link_str.starts_with('#') || link_str.starts_with("/") {
             return LinkLocation::Url(link);
         }
 
-        // Check for local links starting with '/'
-        if link_str.starts_with('/') {
-            link_str = &link_str[1..];
-        }
+        // // Check for local links starting with '/'
+        // if link_str.starts_with('/') {
+        //     link_str = &link_str[1..];
+        // }
+
+        // Take only the first segment for page
+        link_str = link_str.split('#').collect::<Vec<&str>>()[0]
+            .split('/')
+            .collect::<Vec<&str>>()[0];
```

Comment:
Doing `collect()` is an unnecessary allocation, since you only want the first element
```suggestion
        link_str = link_str
            .split('#') // get item before the first #
            .next()
            .expect("Splits always produce at least one item")
            .split('/') // get item before the first /
            .next()
            .expect("Splits always produce at least one item");
```